pilot/pkg/model/push_context.go

```diff
@@ -1353,6 +1353,7 @@ func (ps *PushContext) updateContext(
 		}
 	} else {
 		ps.sidecarIndex.sidecarsByNamespace = oldPushContext.sidecarIndex.sidecarsByNamespace
+		ps.sidecarIndex.rootConfig = oldPushContext.sidecarIndex.rootConfig
```

Comment:
remove this line, test case `TestInitPushContext()` will fail

Reply:
Why not copy whole `sidecarIndex`?

Reply:
- `defaultSidecarMu` is Mutex -- A Mutex must not be copied after first use.
- `computedSidecarsByNamespace` and `gatewayDefaultSidecarsByNamespace` are lazy-ly computed, copying them may improve performance in some cases, e.g. an incoming ADS connection already has a computed entry; but for a fresh new ADS connection which doesn't have Sidecar(namespace/workload level), its SidecarScope still need to be lazy-ly computed.

not copying `rootConfig`  could cause issues, this PR is mainly for bug fix, performance is not concerned -- performance is just like before.

Reply:
defaultSidecarMu is a `*Mutex`, not a `Mutex`. It can be copied.

Not sure why we cannot copy the computed stuff either...? The data within them should still be valid, right?

Reply:
Per my understanding, the computed stuff are still valid, but I didn't copying them because they were not copied before, and not 100% sure copying them is side-effect free.

Reply:
IMO we should copy, it will help performance and make this simpler

Reply:
and if we are not confident its safe we can improve the tests :-)

Reply:
PR updated, PTAL.